docs/source/hyperparameters.rst

```diff
@@ -112,87 +112,87 @@ Often times we train many versions of a model. You might share that model or com
 at which point it is very useful to know how that model was trained (i.e.: what learning rate, neural network, etc...).
 
 Lightning has a few ways of saving that information for you in checkpoints and yaml files. The goal here is to
-improve readability and reproducibility
+improve readability and reproducibility.
 
-1. The first way is to ask lightning to save the values of anything in the __init__ for you to the checkpoint. This also
-makes those values available via `self.hparams`.
+1.  The first way is to ask lightning to save the values of anything in the __init__ for you to the checkpoint. This also
+    makes those values available via `self.hparams`.
 
-.. code-block:: python
+    .. code-block:: python
 
-    class LitMNIST(LightningModule):
+        class LitMNIST(LightningModule):
 
-        def __init__(self, layer_1_dim=128, learning_rate=1e-2, **kwargs):
-            super().__init__()
-            # call this to save (layer_1_dim=128, learning_rate=1e-4) to the checkpoint
-            self.save_hyperparameters()
+            def __init__(self, layer_1_dim=128, learning_rate=1e-2, **kwargs):
+                super().__init__()
+                # call this to save (layer_1_dim=128, learning_rate=1e-4) to the checkpoint
+                self.save_hyperparameters()
 
-            # equivalent
-            self.save_hyperparameters('layer_1_dim', 'learning_rate')
+                # equivalent
+                self.save_hyperparameters('layer_1_dim', 'learning_rate')
 
-            # this now works
-            self.hparams.layer_1_dim
+                # this now works
+                self.hparams.layer_1_dim
 
 
-2. Sometimes your init might have objects or other parameters you might not want to save.
-In that case, choose only a few
+2.  Sometimes your init might have objects or other parameters you might not want to save.
+    In that case, choose only a few
 
-.. code-block:: python
+    .. code-block:: python
 
-    class LitMNIST(LightningModule):
+        class LitMNIST(LightningModule):
 
-        def __init__(self, loss_fx, generator_network, layer_1_dim=128 **kwargs):
-            super().__init__()
-            self.layer_1_dim = layer_1_dim
-            self.loss_fx = loss_fx
+            def __init__(self, loss_fx, generator_network, layer_1_dim=128 **kwargs):
+                super().__init__()
+                self.layer_1_dim = layer_1_dim
+                self.loss_fx = loss_fx
 
-            # call this to save (layer_1_dim=128) to the checkpoint
-            self.save_hyperparameters('layer_1_dim')
+                # call this to save (layer_1_dim=128) to the checkpoint
+                self.save_hyperparameters('layer_1_dim')
```

Comment:
I find this feature a little bit strange. How does it work if I have a list or I want to exclude something ?

Reply:
You can exclude by giving it the names of only what you want, like shown here with layer_1_dim, it will exclude loss_fx, generator_network and **kwargs.

> How does it work if I have a list of key I want to save

You can have a list of names like so: `my_list = ["param1", "param2", ...]` and then
`self.save_hyperparameters(*my_list)`

Reply:
true that case with excluding some could be nice to have...